modules/govuk_postgresql/manifests/wal_e/backup.pp

```diff
@@ -0,0 +1,176 @@
+# == Class: Govuk_postgresql::Wal_e::Backup
+#
+# Creates an offsite backup of a chosen PostgreSQL database
+# to an S3 bucket using WAL-E:
+# https://github.com/wal-e/wal-e
+#
+# === Parameters:
+#
+# [*aws_access_key_id*]
+#   The AWS access ID for the user that is allowed to
+#   access the S3 bucket.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key for the user that is allowed
+#   to access the S3 bucket.
+#
+# [*s3_bucket_url*]
+#   The unique address of the S3 bucket, in the format of:
+#   s3://bucketaddress/directory/foo
+#
+# [*aws_region*]
+#   The AWS region where the specified bucket resides.
+#   Default: eu-west-1
+#
+# [*wale_private_gpg_key*]
+#   The private GPG to import to allow encryption. If present, WAL-E will
+#   will encrypt the backups.
+#
+# [*wale_private_gpg_key_fingerprint*]
+#   The GPG key fingerprint for the above GPG key.
+#
+# [*hour*]
+#   The hour(s) the cron job runs.
```

Comment:
It'd be good to say that the cron job is for the daily base backup (to distinguish it from the incremental backups that happen continuously).